bl/files/files.go

```diff
@@ -13,24 +15,39 @@ type UnknownStruct map[string]interface{}
 func ExtractFilesConfigurations(paths []string, concurrency int) (chan *extractor.FileConfigurations, chan *extractor.InvalidFile) {
 	filesConfigurationsChan := make(chan *extractor.FileConfigurations, concurrency)
 	invalidFilesChan := make(chan *extractor.InvalidFile, concurrency)
+	pathsChan := make(chan string, concurrency)
+
+	go func() {
+		defer close(pathsChan)
+		for _, path := range paths {
+			pathsChan <- path
+		}
+	}()
 
 	go func() {
 		defer func() {
 			close(filesConfigurationsChan)
 			close(invalidFilesChan)
 		}()
 
-		for _, path := range paths {
-
-			configurations, absolutePath, invalidYamlFile := extractor.ExtractConfigurationsFromYamlFile(path)
+		var wg sync.WaitGroup
+		wg.Add(concurrency)
+		for i := 0; i < concurrency; i++ {
+			go func() {
+				defer wg.Done()
+				for path := range pathsChan {
```

Comment:
How about using using a for loop with `path, ok := <- pathChan` to make sure we read form pathsChan until it's both empty and closed?